src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSHostImplementation.cs

```diff
@@ -208,6 +200,154 @@ public static void LoadSatelliteAssembly(byte[] dllBytes)
             AssemblyLoadContext.Default.LoadFromStream(new MemoryStream(dllBytes));
         }
 
+        [UnconditionalSuppressMessage("Trimming", "IL2026", Justification = "Dynamic access from JavaScript")]
+        [UnconditionalSuppressMessage("Trimming", "IL2075", Justification = "Dynamic access from JavaScript")]
+        public static Task<int>? CallEntrypoint(string? mainAssemblyName, string?[]? args, bool waitForDebugger)
+        {
+            if (string.IsNullOrEmpty(mainAssemblyName))
+            {
+                throw new MissingMethodException(SR.MissingManagedEntrypointHandle);
+            }
+
+            Assembly mainAssembly = Assembly.LoadFrom(mainAssemblyName);
+
+            MethodInfo? method = mainAssembly.EntryPoint;
+            if (method == null)
+            {
+                throw new InvalidOperationException(string.Format(SR.CannotResolveManagedEntrypoint, "Main", mainAssemblyName));
+            }
+            if (method.IsSpecialName)
+            {
+                // we are looking for the original async method, rather than for the compiler generated wrapper like <Main>
+                // because we need to yield to browser event loop
+                var type = method.DeclaringType!;
+                var name = method.Name;
+                var asyncName = name + "$";
```

Comment:
We should have a test for every combination of main 
`src\mono\wasm\Wasm.Build.Tests\MainWithArgsTests.cs`

Reply:
I added this as TODO to https://github.com/dotnet/runtime/issues/98366